api/logs/v1/labels_enforcer.go

```diff
@@ -46,6 +46,12 @@ func WithEnforceAuthorizationLabels() func(http.Handler) http.Handler {
 				return
 			}
 
+			if len(matchersInfo.Matchers) == 0 {
+				httperr.PrometheusAPIError(w, "empty authorization label matchers", http.StatusInternalServerError)
```

Comment:
We might do something different than returning such an error. Ideally, we should not run any loki query but directly return an empty result set. (I need to check how to do that...)